x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/credentials_list.tsx

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiBasicTable, EuiButtonIcon, EuiCopy } from '@elastic/eui';
+import { useActions, useValues } from 'kea';
+import { CriteriaWithPagination } from '@elastic/eui/src/components/basic_table/basic_table';
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  CredentialsLogic,
+  ICredentialsLogicActions,
+  ICredentialsLogicValues,
+} from '../credentials_logic';
+import { IApiToken } from '../types';
+import { TOKEN_TYPE_DISPLAY_NAMES } from '../constants';
+import { apiTokenSort } from '../utils/api_token_sort';
+import { getModeDisplayText, getEnginesDisplayText } from '../utils';
+
+export const CredentialsList: React.FC = () => {
+  const { deleteApiKey, fetchCredentials, showCredentialsForm } = useActions(
+    CredentialsLogic
+  ) as ICredentialsLogicActions;
+
+  const { apiTokens, meta } = useValues(CredentialsLogic) as ICredentialsLogicValues;
+
+  const items = apiTokens.slice().sort(apiTokenSort);
```

Comment:
Just an idea / me going all #microperf here - I wonder if we should use [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) here so that we're not running a potentially expensive sort operation every time the component rerenders. We only ever want to re-sort whenever the `apiTokens` array changes, right?

```suggestion
  const items = useMemo(() => apiTokens.slice().sort(apiTokenSort), [apiTokens]);
```

Reply:
Works for me locally (woo!), but we'll have to add a shallow `useMemo` mock to our shallow helpers (darnit). Feel free to leave this one out for now if it's a hassle and I can look at it in a follow-up PR if you like!

Reply:
Good call. I'm happy to make that change.

Reply:
49570d4

Reply:
Awesome! As a heads up, it's possible we'll have to refactor a bit if a component ever uses both `useEffect()` and `useMemo()` and we need to mock both at once 🤔  (I think importing 2 mocks will override the first 1). I'm thinking we'll eventually use a single mock file that mocks all/any `useX` hook we need. Maybe we'll just name it `shallow_react_use_hooks` or something?

Reply:
Actually, wait. Is `credentials_list.test.tsx` even importing the new `shallow_usememo.mock.ts`? If not, then I guess we don't need it - enzyme is just automatically figuring it out for us. You can probably go ahead and delete that new file, sorry 🤦‍♀️

Reply:
Oh wow, there's some weird stuff happening there. Will investigate.

Reply:
Yeah, that was my blunder. It does appear that shallow is running useMemo already. If we need to add something like that again, maybe instead of `shallow_useeffect`, we could do `shallow_reacthooks` or something like that and mock all of the react provided hooks.

Reply:
https://github.com/elastic/kibana/pull/79749/commits/f41a356b47d1d19c4dabf7bc03f9fc595ddd1ecc

Reply:
Sounds great, I love it! Also not your blunder at all, I definitely didn't know Enzyme could handle useMemo either 😁